lib/iris/fileformats/grib/message.py

```diff
@@ -160,7 +164,7 @@ def data(self):
                 shape = (grid_section['numberOfDataPoints'],)
             else:
                 shape = (grid_section['Nj'], grid_section['Ni'])
-            proxy = _DataProxy(shape, np.dtype('f8'), np.nan,
+            proxy = _DataProxy(shape, self.data_dtype, np.nan,
```

Comment:
Seems a bit odd we have two different default `data_dtype` values across the two though...